pkg/apiserver/webhook/nephe_controller_test.go

```diff
@@ -0,0 +1,940 @@
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"antrea.io/nephe/apis/crd/v1alpha1"
+	"antrea.io/nephe/pkg/logging"
+)
+
+var _ = Describe("Webhook", func() {
+	Context("Running Webhook tests for Secret", func() {
+		var (
+			testSecretNamespacedName1 = types.NamespacedName{Namespace: "namespace01", Name: "secret01"}
+			testSecretNamespacedName2 = types.NamespacedName{Namespace: "namespace02", Name: "secret02"}
+			testAccountNamespacedName = types.NamespacedName{Namespace: "namespace01", Name: "account01"}
+			credential                = `{"accessKeyId": "keyId","accessKeySecret": "keySecret"}`
+			credentials               = "credentials"
+			account                   *v1alpha1.CloudProviderAccount
+			s1                        *corev1.Secret
+			encodedS1                 []byte
+			decoder                   *admission.Decoder
+			s1Req                     admission.Request
+			fakeClient                client.WithWatch
+			err                       error
+		)
+
+		BeforeEach(func() {
+			fmt.Println("BeforeEach()")
+			credential = `{"accessKeyId": "keyId","accessKeySecret": "keySecret"}`
+			s1 = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testSecretNamespacedName1.Name,
+					Namespace: testSecretNamespacedName1.Namespace,
+				},
+				Data: map[string][]byte{
+					credentials: []byte(credential),
+				},
+			}
+			encodedS1, _ = json.Marshal(s1)
+			s1Req = admission.Request{
+				AdmissionRequest: v1.AdmissionRequest{
+					UID: "07e52e8d-4513-11e9-a716-42010a800270",
+					Kind: metav1.GroupVersionKind{
+						Group:   "",
+						Version: "corev1",
+						Kind:    "Secret",
+					},
+					Resource: metav1.GroupVersionResource{
+						Group:    "",
+						Version:  "corev1",
+						Resource: "Secrets",
+					},
+					Name:      testSecretNamespacedName1.Name,
+					Namespace: testSecretNamespacedName1.Namespace,
+					Operation: v1.Create,
+					Object: runtime.RawExtension{
+						Raw: encodedS1,
+					},
+				},
+			}
+
+			newScheme := runtime.NewScheme()
+			utilruntime.Must(clientgoscheme.AddToScheme(newScheme))
+			utilruntime.Must(v1alpha1.AddToScheme(newScheme))
+			decoder, err = admission.NewDecoder(newScheme)
+			if err != nil {
+				fmt.Printf("Failed to inject decoder, err %v\n", err)
+			}
+			fakeClient = fake.NewClientBuilder().WithScheme(newScheme).Build()
+		})
+
+		It("Validate Secret create", func() {
+			SecretValidatorTest1 := &SecretValidator{
+				Client: fakeClient,
+				Log:    logging.GetLogger("webhook").WithName("Secret")}
+			err = SecretValidatorTest1.InjectDecoder(decoder)
+			if err != nil {
+				fmt.Printf("Failed to inject decoder, err %v\n", err)
+			}
+			response := SecretValidatorTest1.Handle(context.Background(), s1Req)
+			fmt.Printf("Got admission response %+v\n", response)
```

Comment:
why use standard formatting. Doesnt Ginkgo provide logging.

Reply:
It provides. v2 version of ginkgo gives same options as fmt. 
While the v1 version which we are using, only has one option., as shown below.

```
_, _ = GinkgoWriter.Write([]byte(fmt.Sprintf("Got admission response %+v\n", response)))
```

For now, I have changed log to use the above format, but in future we could upgrade ginkgo to v2.